# A plotting R script produced by the Revigo server at http://revigo.irb.hr/
# If you found Revigo useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );

# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );

# --------------------------------------------------------------------------
# Here is your data from Revigo. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency","plot_X","plot_Y","log_size","value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0000902","cell morphogenesis",0.6354075502089755,-5.68028654404988,-2.6773734854611115,5.411282913017384,-2.889410289700751,0.7708860047112865,0.69769973),
c("GO:0002181","cytoplasmic translation",0.3974763229665376,-6.1691303807045825,3.372016566236975,5.207542813955311,-1.3016384339448903,0.9648705134398015,0.17892117),
c("GO:0006346","DNA methylation-dependent heterochromatin formation",0.018290836776328874,1.9721776321938311,5.44818230916172,3.8705209500127644,-1.5121548798885645,0.7777250927155827,0.82087165),
c("GO:0008618","7-methylguanosine metabolic process",5.6689023831769856E-05,-4.066395360593417,5.179167227624617,1.380211241711606,-1.3872161432802645,0.9246171379238924,0.51078007),
c("GO:0009119","ribonucleoside metabolic process",0.3506733719429951,-4.710016701874202,4.617595079615982,5.153134699238078,-1.3872161432802645,0.8961348436594387,0.88676342),
c("GO:0009266","response to temperature stimulus",0.4441067421784182,3.570245830307615,-5.562126723023542,5.255718634099335,-1.390939450069913,0.8800200274341428,0.33963846),
c("GO:0009653","anatomical structure morphogenesis",1.6026307453462993,-5.319795431499988,-2.539697154440378,5.813062995420341,-3.0915149811213505,0.7802588049321755,0),
c("GO:0009826","unidimensional cell growth",0.016627137163874758,-5.489306372590273,-3.1220903671506246,3.8291107101552946,-2.496209316942819,0.753186959832691,0.84462093),
c("GO:0010035","response to inorganic substance",0.6461143814926801,4.776256027834356,-4.504812274601571,5.418539921951662,-1.3891269996199482,0.828748638426829,0.48279768),
c("GO:0010038","response to metal ion",0.32910689548525657,4.921446077243193,-4.150728746407642,5.125569091658654,-1.6649434805609085,0.8268713240629366,0.79532776),
c("GO:0010043","response to zinc ion",0.10276734124633234,5.108586030282782,-3.9967357713734137,4.620094394032491,-2.9030899869919438,0.8392736807190768,-0),
c("GO:0010184","cytokinin transport",7.394220499796067E-05,-0.8641438037161986,-5.234779142698766,1.4913616938342726,-1.3872161432802645,0.9401937798211344,0.10211233),
c("GO:0010191","mucilage metabolic process",0.001777077660117655,-0.6282554682108406,-7.833066865718365,2.858537197569639,-1.3872161432802645,0.9863152115964754,0.04868117),
c("GO:0010192","mucilage biosynthetic process",0.0017228533764524837,-3.3645958840564143,-7.309374340749358,2.845098040014257,-1.3872161432802645,0.9843137153401839,0.04859334),
c("GO:0022603","regulation of anatomical structure morphogenesis",0.9381565143526257,4.803498235031301,2.936649890035304,5.580505296928462,-2.0357403698031513,0.8562509131382489,0.54797281),
c("GO:0022604","regulation of cell morphogenesis",0.6684621805832305,4.891491339729297,3.1147162231744927,5.433307300026958,-2.0357403698031513,0.8552794378595338,0.85389249),
c("GO:0031047","regulatory ncRNA-mediated gene silencing",0.26322685557224024,3.3210013998311365,4.492070795877011,5.02856311976092,-1.8401321529074333,0.8434508260192995,0.12591571),
c("GO:0031048","regulatory ncRNA-mediated heterochromatin formation",0.03419087559105702,2.147624812774363,5.330697996475193,4.142170386276096,-1.5121548798885645,0.7746761181473388,0.63785474),
c("GO:0032502","developmental process",4.156005433076044,0.7517640125379423,-8.145013696597024,6.2269052215766765,-1.444421927227045,1,-0),
c("GO:0034605","cellular response to heat",0.17319975624705647,3.5074497315913242,-4.96497389528627,4.84678231409796,-1.3872161432802645,0.8292254270145445,0.76046362),
c("GO:0034620","cellular response to unfolded protein",0.15332162680343805,4.2030981711616935,-4.015250944339987,4.793839257525571,-1.3872161432802645,0.756096864476935,0.49956103),
c("GO:0034976","response to endoplasmic reticulum stress",0.252707344541197,3.459891163295046,-4.3662898624337805,5.010850957373923,-1.3872161432802645,0.8404797857433611,0.31837409),
c("GO:0035266","meristem growth",0.0013211007292968974,-5.144797150272465,-3.4329096455969084,2.7299742856995555,-1.3990271043132518,0.7991344804491888,0.72524418),
c("GO:0035966","response to topologically incorrect protein",0.1942264546083099,4.315606591593172,-4.26745759391155,4.896542751228709,-1.3872161432802645,0.8026312592776842,0.61975638),
c("GO:0036498","IRE1-mediated unfolded protein response",0.014633162369096417,4.322053003837767,-2.811769936041569,3.7736401932600256,-1.3872161432802645,0.7341262855683992,0.83313474),
c("GO:0040007","growth",0.14923755234738403,-1.9967810693209291,-7.739464070583327,4.782114147479071,-2.8416375079047502,1,-0),
c("GO:0040029","epigenetic regulation of gene expression",0.14583128143714463,2.156021156790519,4.997936240470042,4.772086889479079,-1.3016384339448903,0.8046103707889662,0.66347235),
c("GO:0042221","response to chemical",4.8560360057130705,4.303758920966944,-5.2551704232982175,6.2945109760284925,-1.6888821573374944,0.8736026540019538,0.28945435),
c("GO:0042278","purine nucleoside metabolic process",0.20156152134410762,-4.5609658579853996,4.539392644864437,4.912641795663937,-1.3872161432802645,0.8988133149167692,0.00554987),
c("GO:0046128","purine ribonucleoside metabolic process",0.14858932568356858,-4.521374452891897,4.755709532676388,4.780223673053241,-1.3872161432802645,0.894579896245917,0.82689536),
c("GO:0048441","petal development",0.00043132952915477056,-5.89963274790644,-1.585679240739801,2.24551266781415,-1.3872161432802645,0.7508431284421919,0.82988649),
c("GO:0048465","corolla development",0.00043132952915477056,-5.623127852721562,-1.4726232752165267,2.24551266781415,-1.3872161432802645,0.7597245670405591,0.82988649),
c("GO:0048589","developmental growth",0.12506091605321748,-5.163242018373431,-3.0394482917000354,4.705359022014651,-2.9281179926938745,0.7549115047154958,0.56360295),
c("GO:0048638","regulation of developmental growth",0.13653921100906757,5.180995987103194,3.149836814264008,4.743494085944266,-1.8401321529074333,0.8346411581785094,0.73349687),
c("GO:0048653","anther development",0.006450225015988769,-5.664274364933928,-1.8508077810471781,3.417969642214737,-1.3872161432802645,0.7165435294863813,0.77458391),
c("GO:0048655","anther wall tapetum morphogenesis",0.00025140349699306625,-6.022188247227987,-1.8071714263221348,2.012837224705172,-1.3872161432802645,0.7234145229458,0.84787201),
c("GO:0048856","anatomical structure development",3.59293800675674,-5.138838217354577,-2.4689325560060467,6.163678879177669,-1.6558042841285652,0.7772026532566816,0.81730353),
c("GO:0051510","regulation of unidimensional cell growth",0.00424921204721614,5.489244529696423,2.882860974990064,3.2367890994092927,-2.020451625295905,0.8568553656739548,0.64340034),
c("GO:0051513","regulation of monopolar cell growth",0.0030119124835835983,5.624657928957535,3.1279182389108655,3.0874264570362855,-2.774690718274137,0.8592695887458767,-0),
c("GO:0052386","cell wall thickening",0.0004830890726533431,-0.19810788760086406,6.461194266248503,2.294466226161593,-1.3872161432802645,0.8998462612855255,0.60237062),
c("GO:0052543","callose deposition in cell wall",0.0004091468676553824,-0.2920769358239754,5.827690102775478,2.2227164711475833,-1.3872161432802645,0.8661772753944313,0.10331555),
c("GO:0055085","transmembrane transport",12.81017399389553,-0.6499705242535809,1.297098401991165,6.715783969524812,-1.5548630312866958,0.9685252027900535,0.18512772),
c("GO:0070828","heterochromatin organization",0.12110500808582658,1.0644535226859506,6.189999572301549,4.691399799098994,-1.3016384339448903,0.8572979042308672,0.19191339),
c("GO:0080170","hydrogen peroxide transmembrane transport",0.00028098037899225057,0.06715045547168355,0.1940337879104162,2.060697840353612,-2.9030899869919438,0.9800208370721707,-0),
c("GO:1900034","regulation of cellular response to heat",0.001885526227447997,7.115378571708191,0.37069095550338776,2.884228769632604,-1.3872161432802645,0.9285902195647286,0.09670477),
c("GO:1900036","positive regulation of cellular response to heat",0.00014788440999592134,7.502549745525755,-0.16680775065588926,1.7853298350107671,-1.3872161432802645,0.9352836678692933,0.50129417),
c("GO:1901068","guanosine-containing compound metabolic process",0.0003006983003250401,-4.150850635616494,4.886229136239333,2.089905111439398,-1.3872161432802645,0.9194489439601324,0.55582167),
c("GO:1905392","plant organ morphogenesis",0.021231271795081108,-5.875512558021953,-2.2573889296237777,3.9352552817840474,-1.3891269996199482,0.7448545220529227,0.50936355));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$log_size <- as.numeric( as.character(one.data$log_size) );
one.data$value <- as.numeric( as.character(one.data$value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = value, size = dispensability), alpha = I(0.6) );
p1 <- p1 + scale_colour_gradientn( colours = c("red", "gray", "blue"), limits = c( min(one.data$value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = dispensability), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) ));
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.2, ];
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/17,max(one.data$plot_X)+one.x_range/12);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/18,max(one.data$plot_Y)+one.y_range/18);


# --------------------------------------------------------------------------
# Output the plot to screen

p1;

nectary_BP <- p1

# Uncomment the line below to also save the plot to a file.
# The file type depends on the extension (default=pdf).

# ggsave("/path_to_your_file/revigo-plot.pdf");

